audbackend/core/artifactory.py

```diff
@@ -1,30 +1,158 @@
 import os
+import typing
 
+import artifactory
 import dohq_artifactory
 
-import audfactory
+import audeer
 
 from audbackend.core import utils
 from audbackend.core.backend import Backend
 
 
+def _artifactory_path(
+        path,
+        username,
+        apikey,
+) -> artifactory.ArtifactoryPath:
+    r"""Authenticate at Artifactory and get path object."""
+
+    return artifactory.ArtifactoryPath(
+        path,
+        auth=(username, apikey),
+    )
+
+
+def _authentication(host) -> typing.Tuple[str, str]:
+    """Look for username and API key."""
+
+    username = os.getenv('ARTIFACTORY_USERNAME', None)
+    api_key = os.getenv('ARTIFACTORY_API_KEY', None)
+    config_file = os.getenv(
+        'ARTIFACTORY_CONFIG_FILE',
+        artifactory.default_config_path,
+    )
+
+    if api_key is None or username is None:
+
+        config = artifactory.read_config(config_file)
+        config_entry = artifactory.get_config_entry(config, host)
+
+        if config_entry is not None:
+            if username is None:
+                username = config_entry['username']
+            if api_key is None:
+                api_key = config_entry['password']
```

Comment:
Those entries might not be part of the config given for that host. It will be safer to change to:

```suggestion
            if username is None:
                username = config_entry.get('username', None)
            if api_key is None:
                api_key = config_entry.get('password', None)
```

Reply:
Good point!

Reply:
I also added a test for it.